### DISCUSSION TITLE
Improve builder tests

### DISCUSSION THREAD

**Discussion by yjaaidi** [jscutlery/semver#26] [Ideas]
1. There should be a `calledTimes(1)` here https://github.com/jscutlery/semver/blob/2fdf2c3b0ff3288e18cdbd3a64b200ae45c05512/packages/semver/src/builders/version/builder.spec.ts#L156 but for some reason, this is called 6 times.
2. `MockBuilderContext` seems to read the filesystem for some reason. We should avoid that.
3. Shouldn't we automock fs etc... with `jest.mock('fs')`?

cc. @edbzn 
Maybe this should be an issue 😅

**Comment by edbzn** [comment]
The number of call comes from the number of tests (6), it seems that the mock isn't cleared after each test, the following line fixe this :

```ts
  afterEach(() => {
    jest.clearAllMocks();
  });
  ```
  
 But I don't really understand why the mock isn't automatically cleared after each test.

**Reply by yjaaidi** [reply]
Actually, it's because we are mocking `fs.readFile` manually. I think it's better to just restore this specific mock to be more explicit.
